libusb/os/darwin_usb.c

```diff
@@ -1666,8 +1668,8 @@ static const struct libusb_interface_descriptor *get_interface_descriptor_by_num
   return NULL;
 }
 
-static enum libusb_error get_endpoints (struct libusb_device_handle *dev_handle, uint8_t iface) {
```

Comment:
Why are some function changed to return int instead of enum?

Reply:
To pacify the C++ compiler, which is more type strict and does not allow such freewheeling conversions between enums and integers.

Reply:
Weird. Does it have to use the C++ compiler. There will be issues if it does because C++ is not a superset of C and many C11 features do not work in C++20.

Reply:
>Does it have to use the C++ compiler.

`-Wthread-safety` partly works in C. The big issue is: https://github.com/llvm/llvm-project/issues/20777

>C++ is not a superset of C 

Indeed, but it mostly is. At least enough for current libusb master.

Again, I'm not wanting to actually switch libusb's build system to C++.  I'm only saying it's handy to do so occasionally, to turn -Wthread-safety on occasionally, like as check before cutting a new release.

Reply:
Hah, they have been trying to figure out what to do for 10 years?! I see some reasonable options but little movement. There is no technical reason why thread analysis should not work with C.

Reply:
>Hah, they have been trying to figure out what to do for 10 years?!

Exactly, alas. So to get the full abilities of -Wthread-safety, we need to build libusb as C++ (or start patching clang). Hence this PR. :)

Reply:
The above 10-year old llvm bug was fixed last week, hurray!

Reply:
No way!?  Nice.

Still, it'll be quite a while before such a compiler is widespread, so I still think we should merge this.